nanoFramework.Device.Bluetooth/BluetoothLEDevice.cs

```diff
@@ -360,48 +400,69 @@ internal GattWriteResult WriteAttributeValueWithResult(ushort attributeHandle, B
         {
             GattCommunicationStatus status = GattCommunicationStatus.Unreachable;
             ushort result = 0;
+            bool securityStarted = false;
+
 
             // Only allow 1 simultaneous call per device connection
             lock (lockObj)
             {
-                result = NativeStartWriteValue(ConnectionHandle, attributeHandle, writeOption == GattWriteOption.WriteWithResponse, value.Data, (ushort)value.Length);
-                if (result == 0)
+                do
                 {
-                    if (writeOption == GattWriteOption.WriteWithResponse)
+                    result = NativeStartWriteValue(ConnectionHandle, attributeHandle, writeOption == GattWriteOption.WriteWithResponse, value.Data, (ushort)value.Length);
```

Comment:
same feedback about the enum and later the case 5

Reply:
added enum